### PR TITLE
SDL2 + X64 Support

### PR DESCRIPTION
**Pull request by simotek**
This pull request adds a new X64 target for 64 bit Linux machines thanks mostly to @NINE-H alongside a working SDL2 port. 

Currently the port is only used on the x64 platform although so far i've also done limited testing for Raspberry Pi and rg35xx which needs a different keymap. They can be added in the near future.

Currently the only known issue is there are race conditions in the RTMidi code so for now its using the DummyMidi backend.

### CODE REVIEW

File: projects/Makefile.PLUSH
Comment (simotek) [inline projects/Makefile.PLUSH:1]: I didn't intentionally include this file intentionally so happy to drop it.

**Comment by simotek** [comment]
For now there are 3 Linux targets:
X64 - SDL2, 64bit
X86 - SDL2, 32bit
DEB- SDL1, 32bit (Should be functionally unchanged)

For now all other targets should be unchanged. They can be tested by modifying the makefile to define `-DSDL2` and changing the LFLAGS

File: projects/Makefile [hunk 1]
```diff
@@ -4,521 +4,443 @@
...
+#---------------------------------------------------------------------------------
+
+LINUXDIRS := \
+	../sources/Adapters/LINUX/Main \
+        ../sources/Adapters/LINUX/System \
+        ../sources/Adapters/Unix/FileSystem \
+	../sources/Adapters/Unix/Process \
+        ../sources/System/Process
+        
+DBDIRS := \
```
Comment (djdiskmachine) [inline projects/Makefile:69]: DEBFILES

**Comment by simotek** [comment]
Ive setup workflows in a branch for testing, currently there are still a couple of issues that need addressing, i'll also attempt to add builds for all 3 of the new targets (X64, X86, rg35xx_stock).

File: projects/Makefile [hunk 2]
```diff
@@ -4,521 +4,443 @@
...
+	../sources/Adapters/DINGOO/Midi \
+	../sources/Adapters/DINGOO/Audio \
+	../sources/Adapters/SDL/Audio \
+	../sources/Adapters/SDL/Process \
+	../sources/Adapters/SDL/GUI \
+	../sources/Adapters/SDL/Timer
+
+RS97DIRS := $(DINGOODIRS)
+BITTBOYDIRS := $(DINGOODIRS)
+MIYOODIRS := $(DINGOODIRS)
```
Comment (djdiskmachine) [inline projects/Makefile:146]: Seems to be missing $(DUMMYMIDIDIRS)

File: projects/Makefile.PLUSH
Comment (igavrilov) [reply projects/Makefile.PLUSH:1]: Suggestion: there is a convention in muOS community to use `SPLUSH24` name, which meant to include entire group of "new" RG35XX devices: S/PLUS/H/2024 (everything except OG).

File: sources/Adapters/SDL2/Input/SDLInput.cpp [hunk 3]
```diff
@@ -0,0 +1,203 @@
+
+#include "SDLInput.h"
+#include "Application/Model/Config.h"
+
+#ifndef PLATOFM_BITTBOY
```
[the remark this answers is not in the document]
Comment (djdiskmachine) [reply sources/Adapters/SDL2/Input/SDLInput.cpp:5]: Didn't realize this was legacy carryover 😅

**Pull request merged**